gryphon/lib/exchange/binance_je.py

```diff
@@ -0,0 +1,188 @@
+"""
+https://github.com/binance-jersey/binance-official-api-docs
+"""
+
+# -*- coding: utf-8 -*-
+import hashlib
+import hmac
+import time
+
+from gryphon.lib.exchange import exceptions
+from gryphon.lib.exchange.exchange_api_wrapper import ExchangeAPIWrapper
+from gryphon.lib.logger import get_logger
+from gryphon.lib.models.exchange import Balance
+from gryphon.lib.money import Money
+
+logger = get_logger(__name__)
+
+
+def binance_signature(params, secret):
+    """Signature of a url query string as required by Binance
+
+    Parameters
+    ----------
+    params : dict
+        query string parameters from a url as given by a request object
+    secret : str
+        binance user secret for signing the query string
+
+    Returns
+    -------
+    str
+    """
+    query_string = ""
+
+    for key in sorted(params.keys()):
+        query_string += key + "=" + str(params[key]) + "&"
+    query_string = query_string[:-1]
+
+    return hmac.new(secret, query_string, hashlib.sha256).hexdigest()
+
+
+class BinanceJeExchange(ExchangeAPIWrapper):
+    currencies = None
+    credentials = None
+    base_url = "https://api.binance.je"
+    endpoints = {
+        "ping": {"req_method": "get", "url": "/api/v1/ping"},
+        "balance": {"req_method": "get", "url": "/api/v3/account"},
+        "prices": {"req_method": "get", "url": "/api/v3/ticker/price"},
+        "open_orders": {"req_method": "get", "url": "/api/v3/openOrders"},
+    }
+
+    def __init__(self, session=None, configuration=None):
+        if self.currencies is None:
+            raise NotImplementedError(
+                "Cannot instantiate BinanceJeExchange. Use a subclass instead."
```

Comment:
It's a gryphon convention to use a trailing comma even when a multiline function call has only a single argument.